redbot/cogs/mod/mutes.py

```diff
@@ -254,22 +255,22 @@ async def guild_mute(self, ctx: commands.Context, user: discord.Member, *, reaso
         audit_reason = get_audit_reason(author, reason)
 
         mute_success = []
-        for channel in guild.channels:
-            success, issue = await self.mute_user(guild, channel, author, user, audit_reason)
-            mute_success.append((success, issue))
-            await asyncio.sleep(0.1)
-        await modlog.create_case(
-            self.bot,
-            guild,
-            ctx.message.created_at,
-            "smute",
-            user,
-            author,
-            reason,
-            until=None,
-            channel=None,
-        )
-        await ctx.send(_("User has been muted in this server."))
+        async with ctx.channel.typing():
+            async for channel in AsyncIter(guild.channels):
```

Comment:
This `AsyncIter` use appears to be wasteful and further slows down how long mutes take. Each channel mute will already yield control of the event loop during communication with Discord

Reply:
Will the mute itself do that or the ``asyncio.sleep(0.1)`` that was here previously. If the mute does it on it's own, I don't see why we were sleeping in the first place here.

Reply:
the .1 sleep that was there previously was similarly unneeded. discord.py already handles any actual rate limit issues this could otherwise cause already, and each set of channel overwrites we want to set involves a network request. Under the hood, when the request is made, a context switch will happen while awaiting a response. There's also nothing preventing this from being done as a gather operation other than that nobody decided to write it this way.